CMakeLists.txt

```diff
@@ -24,35 +24,373 @@
+      - Installation
+        - UNIX apt-get:
+          - apt-get install liblog4cplus-dev
+        - From Source:
+          - https://github.com/log4cplus/log4cplus
+      - Usage
+        Make sure that you set USE_LOG4CPLUS=ON
+        If CMake cannot find your installation location automatically:
+        - Provide Root from environment:
+            export LOG4CPLUS_ROOT=<Installation Root Directory>
+        - Provide Root to CMake:
+            -D LOG4CPLUS_ROOT=<Installation Root Directory>
+        - Provide Locations to CMake:
+            -D LOG4CPLUS_INCLUDEDIR=<Include path>
+            -D LOG4CPLUS_LIBRARYDIR=<Lib path>
+
+    - GLFW: => 3
+      - Simple API for OpenGL development
+      - Installation
+        - UNIX apt-get:
+          - apt-get install libglfw3-dev
+        - Homebrew:
+          - brew install glfw
+        - From Source:
+          - https://www.glfw.org/
+      - Usage
+        - Provide Root from environment:
+            export GLFW3_ROOT=<Installation Root Directory>
+        - Provide Root to CMake:
+            -D GLFW3_ROOT=<Installation Root Directory>
+        - OpenVDB uses GLFW's CMake installation to find and configure
+          the GLFW dependency. You should add GLFW's cmake location to
+          CMAKE_PREFIX_PATH where possible, instead of the above:
+            -D CMAKE_PREFIX_PATH=<GLFW CMake location>
+
+    - Doxygen:
+      - Documentation generation from C++
+      - Installation
+        - UNIX apt-get:
+          - apt-get install doxygen
+        - Homebrew:
+          - brew install doxygen
+        - From Source:
+          - http://www.doxygen.nl/
+
+  Note that various packages have inbuilt CMake module support. See the
+  CMake documentation for more ZLib, Doxygen, OpenGL, Boost and Python
+  controls:
+  https://cmake.org/cmake/help/v3.14/manual/cmake-modules.7.html
+
+  ########################################################################
+
+  OpenVDB Components and their dependencies:
+
+    OpenVDB Core
+      Main CMake: openvdb/CMakeLists.txt
+      Required: IlmBase::Half, TBB::tbb, ZLib, Boost::System,
+        Boost::IOStream
+      Optional: Blosc, Log4cplus, OpenEXR
+
+    OpenVDB Binaries
+      Main CMake: openvdb/cmd/CMakeLists.txt
+      Required: (OpenVDB Core + Dependencies), GLFW3 (vdb_view),
+        OpenGL (vdb_view), Boost::Thread (vdb_view), OpenEXR (vdb_render)
+        IlmBase (vdb_render)
+
+    OpenVDB Python
+      Main CMake: openvdb/python/CMakeLists.txt
+      Required: (OpenVDB Core + Dependencies), Python, Boost::Python
+
+    OpenVDB Unit Tests
+      Main CMake: openvdb/unittest/CMakeLists.txt
+      Required: (OpenVDB Core + Dependencies), CppUnit
+
+  ########################################################################
+
+  The CMake supports building the various components of OpenVDB against
+  a prior installation of OpenVDB.
+
+#]=======================================================================]
```

Comment:
All this information could perhaps be moved somewhere else, but I think its nice to have it here too